"Using the ide" Page [Part 1]

JR1811 opened the pull request:
**Original page:** https://fabric.moddedmc.wiki/getting-started/using-the-ide
**Discussion in the Discord:** https://discord.com/channels/507304429255393322/1208846408552030238/1235450767226437692

## Content

I modified the content slightly to adapt to the new UI, mentioned split sources and added a small recommendation for the decompilers. Besides that, everything else should be mostly the same.

There are no changes in the `reference` directory since this topic doesn't focus on the code but rather on the usage and toolsets of the IDE. There shouldn't be any code logic which needs to be explained further or maintained in the future. They are only examples.

## Problems

The ["Basic Problem solving"](https://fabric.moddedmc.wiki/getting-started/basic-problem-solving) page is linked already, but it's not in the new docs yet. I'll add it to a later pull request asap, if this pull request gets accepted.

Also, VSCode displays the `MD033/no-inline-html: Inline HTML [Element: iframe]` warning on the iframe HTML element at the end of the new page. Is there a better way to implement videos like that?

modmuss50 committed:
Initial commit

modmuss50 committed:
Inital Vitepress setup (#1)

* Inital Vitepress setup

* Actions and change gitignore

* ingore dist

* Clean URLs + srcExclude

* Fix rollup issue, add configured theme.

* Add license.

* Update style.css

* Cleanup + Enable Search

* Minimize css

---------

Co-authored-by: Calum H <contact@mineblock11.dev>

IMB11 committed:
Implement versioning. (#3)

* Implement versioning.

* Sidebar loading from root.

modmuss50 committed:
Add Reference mod (#7)

* Add Reference mod

* 1.20.4 -> latest

modmuss50 committed:
Update homepage (#8)

* Update homepage

* Add docs team to codeowners

modmuss50 committed:
Don't index netlify deployments (#12)

IMB11 committed:
Player Guides (#5)

* Begin on player guides.

* Update FAQ links

* Player installation guides

* Finish windows install java guide.

* Linux install java guide.

* Revert "Update FAQ links"

This reverts commit e612ea4b01f31d9bd6e6ee558a35ceac8ba2d74e.

* Revert index

* Update link for installing Java on MacOS

* Finalize content.

* Rename page title.

* Remove information about inter-loader
compatibility

* Logs + Crash Reports

* Update link to crash reports

* Cleanup of pages, only mention vanilla launcher.

* Add troubleshooting information for mixin crashes

* "Installing Fabric" pages

* Installing mods page.

* Update .gitignore and installing-mods.md

* Fix dead link.

* Updating Fabric guide.

* Apply pull request suggestions.

* Add simple note about opening the installer on MacOS

* Update description in players/index.md

---------

Co-authored-by: modmuss50 <modmuss50@gmail.com>

IMB11 committed:
Sidebar, SEO and Sitemap (#15)

* Sidebar fixes. (Allows hotswapping.)

* Add SEO functionality.

* Sitemap + Icon fixes.

* Remove versioned items from sitemap

* Remove version test.

* Add LanguageTool files to .gitignore

IMB11 committed:
Contribution Guidelines (#16)

IMB11 committed:
Snippet Extension Plugin (#19)

enjarai committed:
Codec Guide (#18)

Superkat32 committed:
"Creating Custom Particles" Page (#20)

* Registering Particles Page

FEAT: Added a new page found within the develop/particles folder. This page guides users through the process of registering a new particle.

* Code Style fix

FIX: Hopefully fixed the incorrect imports

* Migrate to code reference functionality.

* Add extra frontmatter information.

* Disable comment indentation on spotless, conflicts with markdown plugin.

* Revert "Disable comment indentation on spotless, conflicts with markdown plugin."

This reverts commit 62d142205d50f6cb0714052c04167f88bca87a80.

* Solve checkstyle issues.

* Discard changes to package-lock.json

---------

IMB11 committed:
Author System (#21)

IMB11 committed:
General Improvements (#23)

dicedpixels committed:
Events Guide (#22)

IMB11 committed:
Use fork of markdown-snippet that fixes indents. (#25)

dicedpixels committed:
Command Guides (#26)

* add main commands page

* fix formatting using spotless

* requested changes

* Suggestions

* Custom Argument Types

* Refactor BlockPosArgumentType class to remove
unnecessary print statements

* Checkstyle fixes.

* Checkstyle fixes.

* Redirect commands.

* fix checkstyle error

---------

Co-authored-by: Calum <contact@mineblock11.dev>

dicedpixels committed:
Effects and Potions Guides (#28)

IMB11 committed:
Rendering Guides (#24)

IMB11 committed:
Develop Sidebar + Setup spanish for translators (#37)

modmuss50 committed:
Update Crowdin configuration file

FabricMCBot committed:
Update Crowdin configuration file

IMB11 committed:
Warning Banner (#38)

* Develop Sidebar + Setup spanish for translators

* Undo copying of spanish files.

* Update Crowdin configuration file

* Delete crowdin.yml file

* Fix versioning configuration

* Publishing Workflow

* WIP warning.

* Remove VitePress GitHub Pages workflow

* Fix banner logic.

* Fix flash

* Refactor BannerComponent.vue

modmuss50 committed:
Translation tweaks (#40)

* Translation tweaks

* Update

* Dont build l10n branches

modmuss50 committed:
Set workflow concurrency and skip in progress builds

IMB11 committed:
Asset handling fixes. (#43)

* Develop Sidebar + Setup spanish for translators

* Undo copying of spanish files.

* Update Crowdin configuration file

* Delete crowdin.yml file

* Fix versioning configuration

* Publishing Workflow

* WIP warning.

* Remove VitePress GitHub Pages workflow

* Fix banner logic.

* Fix flash

* Refactor BannerComponent.vue

* Fix asset URLs

IMB11 committed:
Dynamic locales + sidebar translations. (#44)

* Develop Sidebar + Setup spanish for translators

* Undo copying of spanish files.

* Update Crowdin configuration file

* Delete crowdin.yml file

* Fix versioning configuration

* Publishing Workflow

* WIP warning.

* Remove VitePress GitHub Pages workflow

* Fix banner logic.

* Fix flash

* Refactor BannerComponent.vue

* Fix asset URLs

* Dynamic sidebar translations + locale loading.

* Update crowdin config.

* Remove old sidebars.json definition from old system + remove console.log

* Root sidebar translations.

* Add missing i18n

FabricMCBot committed:
Update Crowdin configuration file

FabricMCBot committed:
New Crowdin updates (#45)

Co-authored-by: Calum H <contact@mineblock11.dev>

IMB11 committed:
Locale Loading Fix (#46)

modmuss50 committed:
Setup deployment

modmuss50 committed:
Fix publish.yml

modmuss50 committed:
Fix publish

IMB11 committed:
Fix i18n - deadlinks + use Intl instead. (#53)

IMB11 committed:
Locale sidebar URL fixes. (#54)

FabricMCBot committed:
New Crowdin updates (#48)

* New translations index.md (Spanish)

* New translations installing-fabric.md (Spanish)

* New translations installing-mods.md (Spanish)

* New translations updating-fabric.md (Spanish)

* New translations contributing.md (Spanish)

* New translations sidebar_translations.json (Chinese Traditional)

* New translations updating-fabric.md (French)

* New translations index.md (Spanish)

* New translations index.md (Chinese Traditional)

---------

Co-authored-by: Calum H <contact@mineblock11.dev>

IMB11 committed:
Ignore language if index.md is missing. (#55)

Co-authored-by: Fabric Bot <159731069+FabricMCBot@users.noreply.github.com>
Co-authored-by: modmuss <modmuss50@gmail.com>

FabricMCBot committed:
New Crowdin updates (#56)

Benonardo committed:
Small player-facing fixes (#57)

modmuss50 committed:
Replace all "Right Single Quotation Mark"s with an Apostrophe

modmuss50 committed:
More quotation fixes

FabricMCBot committed:
Update Crowdin configuration file

FabricMCBot committed:
New Crowdin updates (#58)

FabricMCBot committed:
Update Crowdin configuration file

Juuxel committed:
index: Fix mroe -> more (#60)

FabricMCBot committed:
New Crowdin updates (#59)

fungoza committed:
small fixes in texts (#62)

FabricMCBot committed:
New Crowdin updates (#64)

* New translations codecs.md (French)

* New translations faq.md (Spanish)

* New translations uploading-logs.md (Spanish)

* New translations windows.md (Russian)

* New translations contributing.md (Chinese Traditional)

* New translations index.md (Chinese Traditional)

* New translations windows.md (Chinese Traditional)

* New translations sidebar_translations.json (Chinese Traditional)

* New translations contributing.md (Portuguese, Brazilian)

* New translations contributing.md (German)

* New translations faq.md (German)

* New translations finding-mods.md (German)

* New translations installing-fabric.md (German)

* New translations installing-mods.md (German)

* New translations crash-reports.md (German)

* New translations uploading-logs.md (German)

* New translations updating-fabric.md (German)

* New translations linux.md (Italian)

* New translations arguments.md (Korean)

* New translations basics.md (Korean)

* New translations suggestions.md (Korean)

* New translations effects.md (Korean)

* New translations potions.md (Korean)

* New translations index.md (Korean)

* New translations sidebar_translations.json (Korean)

* New translations index.md (Ukrainian)

* New translations faq.md (Ukrainian)

* New translations finding-mods.md (Ukrainian)

* New translations index.md (Ukrainian)

* New translations index.md (Vietnamese)

* New translations sidebar_translations.json (Vietnamese)

* New translations contributing.md (Italian)

* New translations codecs.md (Italian)

* New translations index.md (Italian)

* New translations faq.md (Italian)

* New translations finding-mods.md (Italian)

* New translations index.md (Italian)

* New translations installing-fabric.md (Italian)

* New translations windows.md (Italian)

* New translations sidebar_translations.json (Italian)

* New translations contributing.md (Vietnamese)

* New translations index.md (Vietnamese)

* New translations index.md (Vietnamese)

* New translations installing-fabric.md (Vietnamese)

* New translations installing-mods.md (Vietnamese)

* New translations updating-fabric.md (Vietnamese)

* New translations sidebar_translations.json (Vietnamese)

notlin4 committed:
Small Fix (#67)

FabricMCBot committed:
New Crowdin updates (#65)

Superkat32 committed:
Creating Custom Particles Guide Overhaul (#61)

* Creating Custom Particles guide overhaul

All changes are related to the Creating Custom Particles guide.
FEAT: Added things related to the sparkle particle, this includes sentence reworks and code reworks(including file renames)
FEAT: Added an image showcasing the custom particle
FEAT/REFACTOR: Added some tips and information boxes where I felt was fit. Some sections have been replaced to be a tip/info box instead of its own sentence
REFACTOR: Changed the particle's texture from the Fabric logo to a shiny blue sparkle
REFACTOR: Changed the Creating a JSON file and adding textures section to be more easily followed
FIX: JSON is now always properly capitalized when referencing it(not including the actual files)
FIX: Added a period on one of the code comments to be consistent with the rest of the code comments

* Last second image alt text fix

FIX: Somehow my "particle" in "Showcase of the sparkle particle" got lost. I'll also remove "sparkle" just in case that was a style check thing that I didn't notice.

fungoza committed:
Contributing page update (#66)

notlin4 committed:
Grammar fixes (#72)

FabricMCBot committed:
New Crowdin updates (#69)

dicedpixels committed:
Damage Types Guide (#50)

Co-authored-by: Calum H <contact@mineblock11.dev>
Co-authored-by: Fabric Bot <159731069+FabricMCBot@users.noreply.github.com>
Co-authored-by: modmuss <modmuss50@gmail.com>
Co-authored-by: modmuss50 <modmuss50@gmail.com>
Co-authored-by: Benonardo <75318636+Benonardo@users.noreply.github.com>
Co-authored-by: Juuz <6596629+Juuxel@users.noreply.github.com>
Co-authored-by: FurnyGo <68079109+FurnyGo@users.noreply.github.com>
Co-authored-by: notlin4 <121224522+notlin4@users.noreply.github.com>
Co-authored-by: Superkat32 <89557012+Superkat32@users.noreply.github.com>

JR1811 committed:
Sounds Pages (#68)

IMB11 committed:
Fix Codeowners file. (#77)

* Fix locale loading.

* Update CODEOWNERS

* Update CODEOWNERS

IMB11 committed:
Codeowner Fix (#79)

* Fix locale loading.

* Update CODEOWNERS

* Update CODEOWNERS

* Fix

IMB11 committed:
Exclude regional variants of languages on crowdin. (#78)

* Exclude regional variants of languages on crowdin.

* Update excluded target languages in crowdin.yml

FabricMCBot committed:
New Crowdin updates (#74)

Co-authored-by: Calum H <contact@mineblock11.dev>

xpple committed:
Remove incorrect special case for when commands return zero (#76)

* Remove incorrect special case for when commands return zero

* Update develop/commands/basics.md

---------

Co-authored-by: Calum H <hendersoncal117@gmail.com>

Syst3ms committed:
Add recursive codec guide (#75)

Co-authored-by: Calum H <contact@mineblock11.dev>

IMB11 committed:
Fix Incorrect processing of external link in player sidebar. (#81)

IMB11 committed:
Getting Started Section (#47)

FabricMCBot committed:
New Crowdin updates (#80)

kikugie committed:
fix the most minor detail ever, but I'm picky like that (#84)

IMB11 committed:
Add robots.txt (#83)

* robots file

* Update robots.txt

* Update robots.txt

* noindex for versioned pages.

* Remove unused import in seo.ts

FabricMCBot committed:
New Crowdin updates (#82)

FabricMCBot committed:
New Crowdin updates (#85)

* New translations suggestions.md (Chinese Simplified)

* New translations windows.md (Chinese Simplified)

* New translations installing-mods.md (Chinese Simplified)

* New translations uploading-logs.md (Chinese Simplified)

* New translations updating-fabric.md (Chinese Simplified)

* New translations contributing.md (Ukrainian)

* New translations finding-mods.md (Ukrainian)

* New translations installing-fabric.md (Ukrainian)

* New translations installing-mods.md (Ukrainian)

* New translations sidebar_translations.json (Ukrainian)

* New translations suggestions.md (Spanish)

* New translations suggestions.md (German)

* New translations finding-mods.md (Italian)

* New translations linux.md (Italian)

* New translations installing-mods.md (Italian)

* New translations crash-reports.md (Italian)

* New translations updating-fabric.md (Italian)

* New translations sidebar_translations.json (Italian)

* New translations codecs.md (Italian)

* New translations damage-types.md (Italian)

* New translations suggestions.md (Italian)

* New translations sidebar_translations.json (French)

* New translations sidebar_translations.json (Dutch)

* New translations using-sounds.md (Italian)

* New translations creating-a-project.md (Italian)

RooftopJoe committed:
Fix ambiguity regarding the matrix stack (#86)

FabricMCBot committed:
New Crowdin updates (#88)

* New translations introduction-to-fabric-and-modding.md (Italian)

* New translations sidebar_translations.json (Italian)

* New translations effects.md (Italian)

* New translations potions.md (Italian)

* New translations launching-the-game.md (Italian)

* New translations project-structure.md (Italian)

* New translations setting-up-a-development-environment.md (Italian)

* New translations sidebar_translations.json (Chinese Traditional)

fungoza committed:
Fix image description (#92)

FabricMCBot committed:
New Crowdin updates (#91)

Co-authored-by: Calum H <contact@mineblock11.dev>

its-miroma committed:
Update `can-expand` label name (#102)

its-miroma committed:
Correct miscellaneous mistakes (#103)

FabricMCBot committed:
New Crowdin updates (#98)

* New translations codecs.md (Italian)

* New translations sidebar_translations.json (Italian)

* New translations index.md (Turkish)

* New translations suggestions.md (Italian)

* New translations effects.md (Italian)

* New translations potions.md (Italian)

* New translations contributing.md (Italian)

* New translations basics.md (Italian)

* New translations events.md (Italian)

* New translations index.md (Italian)

* New translations basic-concepts.md (Italian)

* New translations draw-context.md (Italian)

* New translations custom-screens.md (Italian)

* New translations windows.md (Italian)

* New translations crash-reports.md (Italian)

* New translations installing-fabric.md (Italian)

* New translations arguments.md (Italian)

* New translations custom-widgets.md (Italian)

* New translations hud.md (Italian)

* New translations creating-particles.md (Italian)

* New translations index.md (Italian)

* New translations faq.md (Italian)

* New translations finding-mods.md (Italian)

* New translations index.md (Italian)

* New translations linux.md (Italian)

* New translations installing-mods.md (Italian)

* New translations uploading-logs.md (Italian)

* New translations updating-fabric.md (Italian)

* New translations damage-types.md (Italian)

* New translations custom.md (Italian)

* New translations using-sounds.md (Italian)

* New translations creating-a-project.md (Italian)

* New translations introduction-to-fabric-and-modding.md (Italian)

* New translations launching-the-game.md (Italian)

* New translations project-structure.md (Italian)

* New translations setting-up-a-development-environment.md (Italian)

* New translations index.md (Russian)

* New translations installing-fabric.md (Russian)

* New translations potions.md (Italian)

* New translations installing-fabric.md (German)

* New translations codecs.md (Italian)

* New translations crash-reports.md (Italian)

* New translations uploading-logs.md (Italian)

* New translations damage-types.md (Italian)

* New translations creating-a-project.md (Italian)

* New translations introduction-to-fabric-and-modding.md (Italian)

* New translations setting-up-a-development-environment.md (Italian)

* New translations contributing.md (Italian)

* New translations sidebar_translations.json (Korean)

* New translations setting-up-a-development-environment.md (Chinese Simplified)

* New translations index.md (Korean)

* New translations codecs.md (Italian)

---------

Co-authored-by: Calum H <contact@mineblock11.dev>

its-miroma committed:
Lint Markdown files, fix headings style (#99)

its-miroma committed:
Update command to clear the effect (#97)

its-miroma committed:
Fix capitalization of "Guidance for expansion" (#106)

its-miroma committed:
Rephrase "Verify if Java is already installed" (#105)

its-miroma committed:
Use relative links (#107)

its-miroma committed:
Use different `ThemeConfig` for each locale (#100)

JR1811 committed:
added image files

JR1811 committed:
added new sidebar page entry

JR1811 committed:
added using-the-ide page

JR1811 committed:
renamed the specific asset folder to match page name

JR1811 committed:
small clean-up and added sidebar translation

JR1811 committed:
changed page link to be relative

IMB11 commented:
It's best if you use the `<video>` tag linking to a video file rather than a YouTube embed.

IMB11 commented:
Hm actually, I thought the video was created by you, but if it's just a Jetbrains video, it can probably just be put into a bullet point list in that section with some other IntellJ IDEA guides

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,236 @@
+---
+title: Using the IDE to the fullest
```
Maybe "IDE Tips and Tricks" instead?

JR1811 committed:
removed video embed and expanded on last section

FabricMCBot committed:
New Crowdin updates (#104)

Co-authored-by: Calum H <contact@mineblock11.dev>

JR1811 committed:
addex extra link for code completion in general

JR1811 committed:
changes title, translation and rearranged last section

IMB11 committed:
Fix search + better locale picking. (#113)

JR1811 commented:
done. Anything else? ^^

its-miroma committed:
Add GitHub workflow for `markdownlint` (#101)

* Add GitHub workflow for `markdownlint`

* Add `MD054` rule

* Add version numbers

* Simplify logic

- Only exclude `ŧranslated` files
- Only lint with `markdownlint-rule-titlecase`

* Lint all files, not only changed files

* Allow all HTML elements

IMB11 committed:
Merge branch 'main' into using-the-ide

IMB11 committed:
Lint fixes (#117)

* Lint fixes.

* Update markdownlint configuration

* Add push trigger for test in markdownlint workflow

* Update .vscode/settings.json and players/updating-fabric.md

* Update markdownlint workflow

IMB11 committed:
Merge branch 'main' into using-the-ide

JR1811 committed:
fixed small typo

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+![Gradle Task](/assets/develop/getting-started/using-the-ide/traversing_01.png)
+
+::: details Recommendation
+A recommendation would be to aim for a Java decompiler, which is as accurate as possible. For example, the [VineFlower](https://github.com/Vineflower/vineflower) option allows for a good experience when debugging the source code of Minecraft.
```
```suggestion
A recommendation would be to use a highly accurate Java decompiler. The VineFlower option, for example, offers a good experience when debugging the source code of Minecraft. More information can be found [here](https://github.com/Vineflower/vineflower).
```

You should probably note that Vineflower is the default compiler in Loom 1.7, this details block probably isnt needed.

JR1811 replied:
yeah, I've written that right before the update notes came xD
ill get rid of that part.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+If you have issues with one of the options, you can always switch later on too, simply by running their Gradle Tasks.
+:::
+
+The Minecraft Source code can be found if you look for `net.minecraft` in the Project Window's External Libraries. If you use the "Split client and common sources" option from the online [Template mod generator](https://fabricmc.net/develop/template/), there will be two sources, as the name suggests. But also other sources of projects, libraries and dependencies, which are imported using the `bui ...
```
Maybe something like:

The Minecraft source code can be found by looking for \`net.minecraft\` in the Project Window's External Libraries. If your project uses split sources from the online \[Template mod generator](https://fabricmc.net/develop/template/), there will be two sources as indicated by the name (client/common). Additionally, other sources of projects, libraries, and dependencies imported via the \`build.gradle\` file will also be available.

I just found this a bit confusing to read.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+
+### Search
+
+Pressing `Shift` twice opens up a Search window. In there you can search for your project's files and classes. When Activating the checkbox `include non-project items`
```
You should put a notice at the top of the page about keybinds being the default keymap, some people switch their IntelliJ IDEA keymaps when they first install it without batting a thought.

Also, it is probably worth using the `<kbd>` tag here for keybinds instead of inline codeblocks?

JR1811 replied:
Well, not all of them are the default ones. E.g. in the comments section, it's explained how to set up custom key binds for that. But yeah, a note for that should be good tho.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+
+![Region collapse](/assets/develop/getting-started/using-the-ide/comments_02.png)
+
+> [!WARNING]
```
Update this to the new callout syntax please.

```
::: warning
...
:::
```

IMB11 replied:
Also, this block isnt really needed, all pages on this site use IDEA.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+![Commit with TODO](/assets/develop/getting-started/using-the-ide/comments_05.png)
+
+> [!WARNING]
+> This feature may not be supported in other editors and IDEs.
```
Same here, not needed, all pages are for IDEA

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+To get started, simply write `/**` above the method or class definition and press enter. IntelliJ will automatically generate lines for the return value
+and the parameters but you can change them however you want. There are many custom functionalities available and you can also use HTML if needed.
+
+Minecraft's `ScreenHandler` class has some examples. To toggle the render view, use the pen button near the line numbers.
```
Should note that any comments within minecraft's source code are from mappings, such as yarn.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+
+### Live Templates
+
+[Youtube IntelliJ IDEA Pro Tips: Live Templates](https://youtu.be/XhCNoN40QTU?si=dGYFr2hY7lPJ6Wge) to generate your custom boilerplate code faster.
```
After #116 we can migrate these videos into actual videoplayers using vidstack.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+
+### PostFix completion
+
+[JetBrains Pagse on Postfix code completion](https://www.jetbrains.com/help/idea/auto-completing-code.html#postfix_completion) to alter code after writing it quickly. Often used examples contain `.not` `.if` `.var` `.null` and `.for`. [Here](https://www.jetbrains.com/help/fleet/postfix-templates-list-java.html) is also a list of all predefined Java Postfix code completions but you can also [creat ...
```
```suggestion
[JetBrains Pagse on Postfix code completion](https://www.jetbrains.com/help/idea/auto-completing-code.html#postfix_completion) to alter code after writing it quickly. Often used examples contain `.not`, `.if`, `.var`, `.null` and `.for`. [Here](https://www.jetbrains.com/help/fleet/postfix-templates-list-java.html) is also a list of all predefined Java Postfix code completions but you can also [create your own](https://www.jetbrains.com/help/idea/auto-completing-code.html#custom-postfix-templates).
```

`.new` is also a commonly used one as well, maybe worth mentioning.

IMB11 commented on develop/getting-started/using-the-ide.md:
```diff
@@ -0,0 +1,246 @@
+This page gives useful bits of information, to speed up and ease the workflow of developers. Incorporate them into yours, to your liking.
+It may take some time to learn and get used to the shortcuts and other options. You can use this page as a reference for that.
+
+> [!NOTE]
```
Migrate this to new callout syntax.

IMB11 commented on develop/getting-started/using-the-ide.md:
````diff
@@ -0,0 +1,246 @@
+
+Now you can highlight the necessary code and use the shortcuts, to comment the section out.
+
+```java
````
What is the purpose of these code blocks, im not sure that they're really needed?

JR1811 replied:
They show how (multiline) comments look like and how to use them.
Lmk if i should remove them.

IMB11 replied:
Yeah I dont think you need a codeblock to emphasise this, most people know what a comment is :stuck_out_tongue_winking_eye:

IMB11 reviewed (changes requested):
I would recommend to use a tool like LanguageTool or an LLM to ensure your sentences are easy to read and grammatically correct. Overall, this is a very useful page and will be beneficial for new mod developers :+1:

IMB11 commented:
Also, make sure you fix the markdownlint issues! I believe the only issues left are the Title Case not being used throughout the page.

IMB11 commented on .vitepress/sidebars/develop.ts:
```diff
@@ -46,6 +46,10 @@ export default [
         link: "/develop/getting-started/launching-the-game"
+      },
+      {
+        text: "develop.gettingStarted.usingIde",
```
nit: May also be worth renaming the translation key and changing the URL to "ide-tips-and-tricks" or something similar.

FabricMCBot committed:
New Crowdin updates (#115)

Co-authored-by: Calum H <contact@mineblock11.dev>

JR1811 committed:
fixed key bindings, readability and other things

JR1811 committed:
moved the renamed page back in

IMB11 committed:
Item pages (#116)

FabricMCBot committed:
New Crowdin updates (#120)

its-miroma committed:
404 page, localisation improvements (#119)

IMB11 committed:
Blocks (#118)

Co-authored-by: modmuss <modmuss50@gmail.com>

FabricMCBot committed:
New Crowdin updates (#124)

Co-authored-by: Calum H <contact@mineblock11.dev>

IMB11 committed:
1.21 (#125)

nddevern committed:
Fix 404 link to custom item groups page and update minecraft version and java version on setting up a development environment page (#127)

Rubyboat1207 committed:
Recursive codecs name change (#128)

megabyte6 committed:
Fix old path (#130)

Change to the correct path for modding in Minecraft version 1.21

IT-Bill committed:
Fix typo in 'GUIDIE' (#132)

cprodhomme committed:
Fix indentation for ModArmorMaterials.java (#136)

cprodhomme committed:
improve order for ModItems tutorial (#135)

I changed the order of the armor items to match the order displayed in the 'Creative' mode:

- Helmet
- Chestplate
- Leggings
- Boots

to improve overall consistency.

Tenneb22 committed:
Hotswapping with JBR (#137)

* Add Hotswapping with a JDK

Adds the method of using the DCVEM or JetBrains JDK to Hotswap classes and methods.

* Add Hotswapping with a JDK

Add the method of using the DCVEM or JetBrain Runtime JDK for hotswapping classes and methods in german.

* Fixed lint issues and updated documentation

* Updated grammar

Co-authored-by: Miroma <136986257+its-miroma@users.noreply.github.com>

---------

Co-authored-by: Miroma <136986257+its-miroma@users.noreply.github.com>

FabricMCBot committed:
New Crowdin updates (#126)

* New translations arguments.md (Chinese Simplified)

* New translations basics.md (Chinese Simplified)

* New translations suggestions.md (Chinese Simplified)

* New translations effects.md (Chinese Simplified)

* New translations faq.md (Chinese Simplified)

* New translations finding-mods.md (Chinese Simplified)

* New translations installing-mods.md (Chinese Simplified)

* New translations crash-reports.md (Chinese Simplified)

* New translations uploading-logs.md (Chinese Simplified)

* New translations updating-fabric.md (Chinese Simplified)

* New translations sidebar_translations.json (Chinese Simplified)

* New translations damage-types.md (Chinese Simplified)

* New translations creating-a-project.md (Chinese Simplified)

* New translations contributing.md (Chinese Simplified)

* New translations website_translations.json (Chinese Simplified)

* New translations blockstates.md (Chinese Simplified)

* New translations first-block.md (Chinese Simplified)

* New translations basic-concepts.md (Chinese Simplified)

* New translations draw-context.md (Chinese Simplified)

* New translations custom-screens.md (Chinese Simplified)

* New translations custom-widgets.md (Chinese Simplified)

* New translations hud.md (Chinese Simplified)

* New translations creating-particles.md (Chinese Simplified)

* New translations index.md (Chinese Simplified)

* New translations index.md (Chinese Traditional)

* New translations installing-fabric.md (Chinese Simplified)

* New translations sidebar_translations.json (Chinese Traditional)

* New translations custom.md (Chinese Simplified)

* New translations using-sounds.md (Chinese Simplified)

* New translations setting-up-a-development-environment.md (Chinese Simplified)

* New translations website_translations.json (Chinese Traditional)

* New translations codecs.md (Chinese Simplified)

* New translations events.md (Chinese Simplified)

* New translations potions.md (Chinese Simplified)

* New translations linux.md (Chinese Simplified)

* New translations windows.md (Chinese Simplified)

* New translations custom.md (Chinese Simplified)

* New translations using-sounds.md (Chinese Simplified)

* New translations first-item.md (Chinese Simplified)

* New translations text-and-translations.md (Chinese Simplified)

* New translations basics.md (Russian)

* New translations effects.md (Russian)

* New translations creating-a-project.md (Chinese Traditional)

* New translations introduction-to-fabric-and-modding.md (Russian)

* New translations launching-the-game.md (Russian)

* New translations introduction-to-fabric-and-modding.md (Chinese Traditional)

* New translations launching-the-game.md (Chinese Traditional)

* New translations project-structure.md (Chinese Traditional)

* New translations setting-up-a-development-environment.md (Chinese Traditional)

* New translations custom-item-groups.md (Russian)

* New translations custom-item-interactions.md (Russian)

* New translations custom-armor.md (Chinese Simplified)

* New translations custom-item-groups.md (Chinese Simplified)

* New translations index.md (Russian)

* New translations basics.md (Russian)

* New translations faq.md (Korean)

* New translations sidebar_translations.json (Russian)

* New translations custom.md (Russian)

* New translations project-structure.md (Russian)

* New translations index.md (Korean)

* New translations website_translations.json (Korean)

* New translations website_translations.json (Russian)

* New translations effects.md (Italian)

* New translations sidebar_translations.json (Italian)

* New translations using-sounds.md (Italian)

* New translations effects.md (Portuguese, Brazilian)

* New translations codecs.md (Italian)

* New translations arguments.md (Italian)

* New translations basics.md (Italian)

* New translations suggestions.md (Italian)

* New translations events.md (Italian)

* New translations potions.md (Italian)

* New translations basic-concepts.md (Italian)

* New translations draw-context.md (Italian)

* New translations hud.md (Italian)

* New translations creating-particles.md (Italian)

* New translations index.md (Italian)

* New translations faq.md (Italian)

* New translations finding-mods.md (Italian)

* New translations installing-fabric.md (Italian)

* New translations linux.md (Italian)

* New translations windows.md (Italian)

* New translations installing-mods.md (Italian)

* New translations crash-reports.md (Italian)

* New translations uploading-logs.md (Italian)

* New translations updating-fabric.md (Italian)

* New translations damage-types.md (Italian)

* New translations custom.md (Italian)

* New translations introduction-to-fabric-and-modding.md (Italian)

* New translations setting-up-a-development-environment.md (Italian)

* New translations contributing.md (Italian)

* New translations damage-types.md (Portuguese, Brazilian)

---------

Co-authored-by: Calum H <contact@mineblock11.dev>

JaaiDead committed:
Potions Minor Code Reference Change (#140)

* Potions Minor Code Change

* edit update

modmuss50 commented:
I think this page is really good, I learnt some stuff too! However I dont think its suitable for the "Getting Started" section, most of this will likely overwhelm someone who is just getting started. I would suggest moving it to the "Misc" group alongside codecs.

JR1811 committed:
Merge remote-tracking branch 'upstream/main' into using-the-ide

JR1811 committed:
fixed header linting issues

szdytom committed:
Fix a code link typo (#143)

JR1811 committed:
fixed more markdown linting issues and replaced most external links with videos

JR1811 committed:
Merge branch 'FabricMC:main' into using-the-ide

JR1811 committed:
fixed idea specific directories in gitignore
